### PR TITLE
RD-4403: Creating secrets on terraform modal submission

### PR DESCRIPTION
**Pull request by jawamartus**
## Description
RD-4403

## Checklist

- [x] My code follows the [UI Code Style](https://cloudifysource.atlassian.net/l/c/x8fq902p).
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires any update in the [documentation](https://cloudifysource.atlassian.net/l/c/4XKtPocy).
- [x] I added proper labels to this PR.

## Tests

 

Blueprint spec single test: [Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/2717/)

[Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/2716/) - new

[Build Status](https://jenkins.cloudify.co/job/Stage-UI-System-Test/2714/)
## Documentation

N/A

### CODE REVIEW

File: widgets/common/src/terraformModal/TerraformModal.tsx [hunk 1]
```diff
@@ -308,6 +308,21 @@ export default function TerraformModal({
...
+            const secretActions = new SecretActions(toolbox);
+            const { defaultVisibility } = Consts;
+            const allSecretVariables: Variable[] = [...variables, ...environment].filter(
+                variable => variable.source === 'secret'
+            );
+
+            await allSecretVariables.forEach(async secretVariable => {
+                // add secret if not exist
+                await secretActions.doGet(secretVariable.name).catch(async e => {
+                    await secretActions.doCreate(secretVariable.name, secretVariable.value, defaultVisibility, false);
```
Comment (kubama) [inline widgets/common/src/terraformModal/TerraformModal.tsx:321]: I think performing `doGet` is not necessary - if secret exists `doCreate` could just silently fail. The effect would be the same but we would save on the number of requests.

Comment (jawamartus) [reply widgets/common/src/terraformModal/TerraformModal.tsx:321]: I prefer to keep it this way. I don't want to override the secrets. Now there is no documentation about using creating secrets with existing key. Even if now it returns error I cannot predict that it wouldn't change in the future. https://docs.cloudify.co/api/v3.1/#create-secret

**Comment by jawamartus** [comment]
I fixed system tests.

**Review by kubama — DISMISSED** [review]

**Comment by jawamartus** [comment]
By mistake I committed "it.only". Fixed. Asking for rereview.

**Review by kubama — APPROVED** [review]

**Pull request merged**